### DISCUSSION TITLE
Change `"ctrl+shift+right"` to match RStudio

### DISCUSSION THREAD

**Discussion by TimTaylor** [posit-dev/positron#5276] [Feedback]
In RStudio pressing `"ctrl+shift+right"` is my usual way of removing white space between the end of one line and the start of commands lower down. In positron this selects not just whitespace but also goes to the end of the next word. Would you consider adding something similar to the following to the RStudio keymappings:

```
[{
    "key": "ctrl+shift+right",
    "command": "cursorWordStartRightSelect",
    "when": "textInputFocus"
}]
```

EDIT - this still doesn't quite work but now in a different way. It grabs white-space past the end of a word. I *think* I'm after something that doesn't ignore whitespace :thinking: 

Note that [this issue](https://github.com/microsoft/vscode/issues/208487) suggests it is a Linux specific behaviour but I'm sure I've use the same shortcut in other OS so :man_shrugging:

**Comment by TimTaylor** [comment]
maybe `WordPart` is what matches RStudio?
```json
[{
    "key": "ctrl+shift+right",
    "command": "cursorWordPartRightSelect",
    "when": "textInputFocus"
}]
```

**Reply by juliasilge** [reply]
Thanks for sharing this feedback! 

If your specific pain point is around trailing whitespace, as a workaround for now you may want to set opt in to this setting:

![Screenshot 2024-11-07 at 4 12 37 PM](https://github.com/user-attachments/assets/fe5ce2f1-0e19-4522-8906-b5a83c6539d0)

In JSON this will look like:

```json
{
    "files.trimTrailingWhitespace": true
}
```

**Reply by TimTaylor** [reply]
Thank you but it's not about that, but code navigation across words. By default `ctrl+shift+right` and `ctrl+shift+left` behave differently to RStudio irrespective of whether the RStudio keymappings are enabled or not.

Setting `"cursorWordPartRightSelect"` and `"cursorWordPartLeftSelect"` gives comparable behaviour AFAICT.

**Reply by juliasilge** [reply]
Ah, thanks for the clarification! Have you set this as a [custom keybinding](https://code.visualstudio.com/docs/getstarted/keybindings) for yourself to experiment with? It would be interesting to know if after some time using this, you find it close enough to the RStudio behavior.
